### PR TITLE
Pigeon Bullet Proofing

### PR DESCRIPTION
**Pull request by Lockscout**

### CODE REVIEW

**Review by nrgill28 — APPROVED** [review]

**Review by LaraLim — APPROVED** [review]

File: src/main/java/frc/robot/subsystems/DriveBase.java [hunk 1]
```diff
@@ -51,10 +51,16 @@ private DriveBase() {
...
+        talon5plyboy = new WPI_TalonSRX((Config.TALON_5_PLYBOY));
+
         robotDriveBase = new DifferentialDrive(leftFrontTalon, rightFrontTalon);
-        _pidgey = new PigeonIMU (Config.robotSpecific(null, null, null, leftFrontTalon, leftRearTalon));
 
-        _pidgey.setFusedHeading(0.0, 30);
+
+        var pigeonTalon = Config.robotSpecific(null, null, rightRearTalon, leftFrontTalon, leftRearTalon, talon5plyboy);
+        if(pigeonTalon != null){
+            _pidgey = new PigeonIMU (pigeonTalon);
```
Comment (nicojopr) [inline src/main/java/frc/robot/subsystems/DriveBase.java:61]: do we need to check _pidgey is non-null as well before calling setFusedHeading?

File: src/main/java/frc/robot/commands/DrivetrainPIDTurnDelta.java [hunk 2]
```diff
@@ -58,6 +58,14 @@ public DrivetrainPIDTurnDelta(double deltaDegree, double forwardSpeed) {
         double currentAngle = drivebase.getCurrentAngle();
         logger.addHandler(Config.logFileHandler);
 
+        if(_pidgey == null){
+
```
Comment (nicojopr) [inline src/main/java/frc/robot/commands/DrivetrainPIDTurnDelta.java:62]: What is best practice in the robot code if components are unavailable? Log it at least in my opinion.

File: src/main/java/frc/robot/subsystems/DriveBase.java (hunk 1 above)
Comment (Lockscout) [reply src/main/java/frc/robot/subsystems/DriveBase.java:61]: From what we have tested, its not needed to detect that for the setFusedHeading

File: src/main/java/frc/robot/commands/DrivetrainPIDTurnDelta.java (hunk 2 above)
Comment (Lockscout) [reply src/main/java/frc/robot/commands/DrivetrainPIDTurnDelta.java:62]: Thats true, Ill look into it

**Pull request merged**